core/domain/user_services_test.py

```diff
@@ -1363,11 +1429,102 @@ def test_validate_creator_dashboard_display_pref(self):
     def test_guest_has_not_fully_registered(self):
         self.assertFalse(user_services.has_fully_registered_account(None))
 
-    def test_cannot_create_new_user_with_existing_user_id(self):
-        with self.assertRaisesRegexp(
-            Exception, 'User %s already exists.' % self.owner_id
-        ):
-            user_services.create_new_user(self.owner_id, self.OWNER_EMAIL)
+    def test_save_user_auth_for_user_auth_enabled_creates_new_auth_entry(self):
+        with self.swap(feconf, 'ENABLE_USER_AUTH_MODEL', True):
```

Comment:
This should happen even when the flag is false (See my above comment about how we dont need to flag protect creating the user )